lib/reverse_sentence.rb

```diff
@@ -1,6 +1,86 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
-def reverse_sentence(my_sentence)
-  raise NotImplementedError
-end
+# Time complexity : O(nˆ3) 3 nested loops, cubic
+# Space complexity: O(1) in-place
+# In-place algorithm is an algorithm which transforms input using no auxiliary data structure
+# A small amount of extra storage space is allowed for auxiliary variables
+# The input is usually overwritten by the output as the algorithm executes
+# In-place algorithm updates input sequence only through replacement or swapping of elements
+# N (or n) is the number of characters in the input String
+# Input: "Yoda is cool and great ", Output: "great and cool is Yoda"
+
+def reverse_sentence(str)
```

Comment:
Actually I think your method is O(n) time complexity because your inner loops affect the runtime of the outer loop.  They shift `left_end` over  

Do notice that you have code to reverse sections of the string (and the whole string) twice here.  Could you abstract that into a helper method?